nflow-engine/src/main/java/com/nitorcreations/nflow/engine/internal/dao/ArchiveDao.java

```diff
@@ -0,0 +1,110 @@
+package com.nitorcreations.nflow.engine.internal.dao;
+
+import static com.nitorcreations.nflow.engine.internal.storage.db.DatabaseConfiguration.NFLOW_DATABASE_INITIALIZER;
+
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.util.LinkedList;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.apache.commons.lang3.StringUtils;
+import org.joda.time.DateTime;
+import org.springframework.context.annotation.DependsOn;
+import org.springframework.dao.DataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.nitorcreations.nflow.engine.internal.config.NFlow;
+
+@Named
+@DependsOn(NFLOW_DATABASE_INITIALIZER)
+public class ArchiveDao {
+  private JdbcTemplate jdbc;
+  private TableMetadataChecker tableMetadataChecker;
+
+  @Inject
+  public void setJdbcTemplate(@NFlow JdbcTemplate jdbcTemplate) {
```

Comment:
JdbcTemplate is injected via a setter elsewhere. And I recall that there was some reason for that?